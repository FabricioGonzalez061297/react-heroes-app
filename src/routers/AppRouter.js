import React, { useContext } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

export const AppRouter = () => {

    const {user:{logged:isAuthenticated} } = useContext(AuthContext);

    return (
       <Router>
           <div>

                <Switch>
                    <PublicRoutes  isAuthenticated={isAuthenticated}  exact path="/login" component={LoginScreen} />
                    <PrivateRoutes isAuthenticated={isAuthenticated} path="/" component={DashboardRoutes} />
                </Switch>
           </div>
       </Router>
    )
}
