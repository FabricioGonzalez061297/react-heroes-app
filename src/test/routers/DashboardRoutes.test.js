import { mount, shallow } from "enzyme";
import React from 'react'
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";


describe('Pruebas en <DashboardRoutes/>', () => {

    const authContext={
        user:{name:'fabricio',logged:true},
        dispatch:jest.fn()
    }

    test('Debe mostrarse correctamente', () => {
            const wrapper= mount(
                <AuthContext.Provider value={authContext} >
                    <MemoryRouter>
                    <DashboardRoutes />
                    </MemoryRouter>
                </AuthContext.Provider>
            );
            
            expect(wrapper).toMatchSnapshot();
            expect(wrapper.find('.text-info').text().trim()).toBe('fabricio');
    });
    
})
