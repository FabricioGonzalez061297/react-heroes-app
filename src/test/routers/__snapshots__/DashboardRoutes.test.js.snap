// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes/> Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": ".5rem",
        }
      }
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <ul
      className="navbar-nav "
    >
      <span
        className="nav-item nav-link text-info "
      >
         
        fabricio
         
      </span>
      <button
        className="nav-item nav-link btn "
        onClick={[Function]}
        style={
          Object {
            "marginRight": ".5rem",
          }
        }
        type="button"
      >
        Logout
      </button>
    </ul>
  </nav>,
  <div
    className="container mt-4"
  >
    <div>
      <h2>
        Marrvel screen
      </h2>
      <div
        className="row row-cols-2 row-cols-md-4 g-4 animate__animated  animate__fadeIn "
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Spider Man"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-spider.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Spider Man
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Peter Parker
                 
              </p>
              <a
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Captain America"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-captain.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Captain America
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Steve Rogers
                 
              </p>
              <a
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Iron Man"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-iron.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Iron Man
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Tony Stark
                 
              </p>
              <a
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Thor"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-thor.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Thor
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Thor Odinson
                 
              </p>
              <a
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Hulk"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-hulk.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Hulk
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Banner
                 
              </p>
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Wolverine"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-wolverine.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Wolverine
                 
              </h5>
              <p
                className="card-text"
              >
                 
                James Howlett
                 
              </p>
              <a
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Daredevil"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-daredevil.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Daredevil
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Matthew Michael Murdock
                 
              </p>
              <a
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Hawkeye"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-hawkeye.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Hawkeye
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Clinton Francis Barton
                 
              </p>
              <a
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Cyclops"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-cyclops.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Cyclops
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Scott Summers
                 
              </p>
              <a
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <img
              alt="Silver Surfer"
              className="card-img-top img-thumbnail "
              src="assets/heroes/marvel-silver.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Silver Surfer
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Norrin Radd
                 
              </p>
              <a
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Más..
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
